posts/2017-09-27-designing-the-event-gateway.md

```diff
@@ -1,55 +1,63 @@
+To keep us on track, we decided to establish four core guiding principles. It doesn't mean that they won't change in the future—they surely will—but we wanted to have something that would give us a jump start during the implementation phase and carry throughout the project development.
 
 Those guiding principles were:
 
-- **Simplicity** - Keep the number of new concepts low. We didn't want to rename existing concepts just for the sake of it. There would be only events, functions, and subscriptions - as easy as possible. In terms of operational simplicity, our goal was to make it easy to run the event gateway locally, during development, while also making it easy to run, manage and scale in production environments.
+- **Simplicity** - Keep the number of new concepts low. We didn't want to rename existing concepts just for the sake of it. There would be only events, functions, and subscriptions—as easy as possible. In terms of operational simplicity, our goal was to make it easy to run the event gateway locally during development, while also making it easy to run, manage and scale in production environments.
 - **Cross-cloud** - Same as with the Serverless Framework, the event gateway had to provide a seamless user experience, no matter where the user deploys their functions or where they want to deploy the Event Gateway itself. In the latter case, we also needed to support on-premise deployments.
 - **Event-driven** - We strongly believed that event-driven was the right approach for building software systems. The Event Gateway should not only enable developers to build them, but should also follow this paradigm internally.
 - **Optimized for fast delivery** - The main goal of the project was event delivery. We wanted to make it instant.
 
-## Architectural Choices
+# Architectural Choices
 
-Having all those principles in mind we made a few explicit, architectural choices that drove the implementation.
+With those principles in mind we made a few explicit, architectural choices that drove the implementation.
 
-### Stateless
+## Stateless
 
-The event gateway was designed to be a stateless service backed by an external key-value store. This makes it easy to operate and reason about.
+The event gateway was designed to be a stateless service backed by an external key-value store. This was to make it easy to operate and reason about.
 
-Assuming that we want to build a horizontally scalable system, making the event gateway a stateful service meant that we would need to implement yet another distributed database. As that did not fit with our core values (remember: Simplicity!), we deciced to avoid it.
+Assuming that we want to build a horizontally scalable system, making the event gateway a stateful service meant that we'd need to implement yet another distributed database. As that did not fit with our core values (remember: Simplicity!), we decided to avoid it.
```

Comment:
> (remember: Simplicity!)

I would remove it. It might be misleading. It's not only about simplicity, it's more about business values that the company provide.